### PR TITLE
add github action

### PR DESCRIPTION
**Pull request by Roguelazer**
see internal task T28704 for details

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -1,3 +1,10 @@
+name: 'CI'
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
```
Comment (Justintime50) [inline .github/workflows/test.yaml:6]: Is this correct `yml` syntax here? Leaving the colon without anything inside (beneath) it?

Comment (Roguelazer) [reply .github/workflows/test.yaml:6]: it's what the github docs say to do. I believe it's equivalent to saying `pull_request: ~` and should deserialize to

```
{"pull_request": None}
```

Comment (Justintime50) [reply .github/workflows/test.yaml:6]: Perfect! Thanks

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - uses: actions/checkout@v2
+      - name: set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.pythonversion }}
+      - name: install dependencies
+        run: |
+          python -m pip install six requests flake8
+          python -m pip install -r requirements-tests.txt
+          python -m pip install -e .
```
Comment (Justintime50) [inline .github/workflows/test.yaml:24]: Won't `python -m pip install -e .` also be installing the requirements from `setup.py` and so installing `six` and `requests` separate above would be redundant?

Comment (Roguelazer) [reply .github/workflows/test.yaml:24]: probably. I was just translating from the travis config, which copied that from the circle config, which got it from gods know where

**Review by Justintime50 — APPROVED** [review]
LGTM, would love to get flake8 into our test requirements so it's ready to use whenever including locally.

**Comment by Justintime50** [comment]
Ah, can we also add the GitHub Actions README badge and swap it for the Travis badge?

**Review by Justintime50 — APPROVED** [review]
LGTM

**Comment by Justintime50** [comment]
Looks like we'll need to remove Travis CI checks in the settings of the repo before this goes green.

**Pull request merged**